ingest/validation/validate_metadata.py

```diff
@@ -204,61 +451,176 @@ def validate_cells_unique(metadata):
+            metadata.store_validation_issue('warn', 'ontology', error_msg, [cell_id])
+
+    metadata.ontology[property_name][(row[property_name], row[ontology_label])].append(
+        cell_id
+    )
+    return row
+
+
+def collect_cell_for_ontology(
+    property_name, row, metadata, convention, array, required
+):
     """Collect ontology info for a single metadatum into CellMetadata.ontology dictionary
+    if ontology_label not provided for optional metadata, label is looked up
+    and added to BigQuery json to populate metadata backend
+    Missing ontology_label for required metadata is not inserted, should fail validation
     """
-    if metadatum.endswith('__unit'):
-        ontology_label = metadatum + '_label'
+
+    if property_name.endswith('__unit'):
+        ontology_label = property_name + '_label'
     else:
-        ontology_label = metadatum + '__ontology_label'
-    cell_id = row['CellID']
-    if array:
-        # Catch unusual edge case where ontology_label column provided but empty
-        # np.isnan does not support string values and generates a TypeError
-        try:
-            np.isnan(row[ontology_label])
-            for id in row[metadatum]:
-                metadata.ontology[metadatum][(id)].append(cell_id)
-        except (TypeError, KeyError):
-            # Not all ontology data will have user-provided ontology_label
-            # where provided, collect to provide cross-validation
-            try:
-                ontology_dict = dict(zip(row[metadatum], row[ontology_label]))
-                for id, label in ontology_dict.items():
-                    metadata.ontology[metadatum][(id, label)].append(cell_id)
-            # where ontology_label not provided, collect ontology id for validation
-            except (TypeError, KeyError):
-                for id in row[metadatum]:
-                    metadata.ontology[metadatum][(id)].append(cell_id)
+        ontology_label = property_name + '__ontology_label'
+    updated_row = copy.deepcopy(row)
+    cell_id = updated_row['CellID']
+
+    # for case where they've omitted a column altogether or left it blank, add a blank entry
+    # the check for != is because pandas converts blank to nan.
+    # See https://stackoverflow.com/questions/944700/how-can-i-check-for-nan-values
+    if (
+        ontology_label not in updated_row
+        or updated_row[ontology_label] != updated_row[ontology_label]
+    ):
+        if array:
+            updated_row[ontology_label] = []
+        else:
+            updated_row[ontology_label] = ''
+
+    # !!!property_name checking is probably not needed
+    if (
+        property_name not in updated_row
+        or updated_row[property_name] != updated_row[property_name]
+    ):
+        if array:
+            updated_row[property_name] = []
+        else:
+            updated_row[property_name] = ''
```

Comment:
I've updated the function value_is_nan(x) to use math.isnan(x). value_is_nan(x) will properly handle the case where X is not a float and return False instead of a ValueError. I believe this will improve readability and still handle the edge case that exists for using math.isnan. Note: choosing here to use math.isnan(x) over np.isnan(x) to reduce the number of 3rd party library dependencies and increase the potential for portability.